docs/zk-passport/guide-on-chain-verification.mdx

```diff
@@ -0,0 +1,186 @@
+---
+description: Use ZK Passport to verify users on-chain without revealing private data. Learn how to implement the verification process step-by-step.
+sidebar_label: "Guide: on-chain verification"
+---
+
+import OutLink from "@site/src/components/OutLink";
+import IdealImage from '@site/src/components/IdealImage';
+
+# Guide: Setting up on-chain user verification with ZK Passport
+
+This tutorial walks you through the process of setting up on-chain verification of ZK Passport proofs on-chain. By the end of this tutorial, you will have:
+
+1. Ensure that you have all the necessary components in place to verify ZK Passport proofs on your chain.
+1. Deployed the **Verificator** contract on your chain to handle proof verification.
+1. Integrated verification logic in your own contract to validate proof public signals.
+
+>TODO: add step for setting up the front-end with our new SDK
+
+## Prerequisites
+
+- In case you are using an EVM-compatible chain other than Rarimo's L2, you'll need to to set up the ZK Passport Registry state replication as described in [Setting up cross-chain verification](/zk-passport/guide-setting-up-cross-chain-replication).
+- Access to the [`verificator-svc`](https://github.com/rarimo) service for retrieving proof parameters. You can use the public instance of `verificator-svc` <OutLink href="https://api.app.rarime.com/">https://api.app.rarime.com/</OutLink>  for testing, but it's recommended to deploy your own instance for production use, as described in [Setting up verificator-svc](/zk-passport/guide-setting-up-verificator-svc).
+
+## Step 1: Deploy the Verificator smart contract
+
+>TODO: put this contract into an npm package?
+>TODO: guide to customizing the query params
+
+The **Verificator** contract is responsible for verifying the ZK proofs on-chain. This contract often comes with a precompiled ZK circuit.
+
+1. Obtain the `Verificator` contract <OutLink href="https://github.com/rarimo/passport-voting-contracts/blob/4cccd0ca8613330c4af68d8e0f7ca46bd3b68ac9/contracts/voting/verifiers/BioPassportVotingVerifier.sol">source code</OutLink> and add it to your project.
```

Comment:
let's call smart contract `verifier` to get rid of potential mess with verificator-svc